py/desispec/scripts/coadd_spectra.py

```diff
@@ -54,12 +55,46 @@ def main(args=None):
     if len(args.infile) == 0:
         log.critical("You must specify input files")
         return 12
-    
-    log.info("reading spectra ...")
-        
-    if len(args.infile) == 1:
+
+    log.info("reading input ...")
+
+    # inspect headers
+    input_is_frames = False
+    input_is_spectra = False
+    for filename in args.infile :
+        ifile = fitsio.FITS(filename)
+        head=ifile[0].read_header()
+        identified = False
+        if "EXTNAME" in head and head["EXTNAME"]=="FLUX" :
+            print(filename,"is a frame")
+            input_is_frames = True
+            identified = True
+            ifile.close()
+            continue
+        for hdu in ifile :
+            head=hdu.read_header()
+            if "EXTNAME" in head and head["EXTNAME"].find("_FLUX")>=0 :
+                print(filename,"is a spectra")
+                input_is_spectra = True
+                identified = True
+                break
+        ifile.close()
+        if not identified :
+            log.error("{} not identified as frame of spectra file".format(filename))
+            sys.exit(1)
+
+
+    if input_is_frames and input_is_spectra :
+        log.error("cannot combine input spectra and frames")
+        sys.exit(1)
+
+
+    if input_is_spectra :
         spectra = read_spectra(args.infile[0])
-    else:
+        for filename in args.infile[1:] :
+            log.info("append {}".format(filename))
+            spectra.update(read_spectra(filename))
```

Comment:
We should consider merging #1107 and then using `desispec.spectra.stack` instead which is way faster than `Spectra.update`, albeit with the restriction that the input spectra have to be homogeneous (e.g. they all have the same cameras).

Reply:
ok

Reply:
to avoid rebasing confusion between two different open PRs, we can address this optimization later.